### PR TITLE
picom-pijulius: 8.2-unstable-2024-04-30 -> 8.2-unstable-2024-06-13

### PR DESCRIPTION
**Pull request by YvesStraten**
## Description of changes
This commit updates picom-pijulius, and adds support for `passthru.updateScript`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/pi/picom-pijulius/package.nix [hunk 1]
```diff
@@ -2,17 +2,18 @@
...
   lib,
   fetchFromGitHub,
-  pcre
+  pcre,
+  nix-update-script
 }:
 picom.overrideAttrs (previousAttrs: {
   pname = "picom-pijulius";
-  version = "8.2-unstable-2024-04-30";
+  version = "Next-unstable-2024-06-06";
```
Comment (JohnRTitor) [inline pkgs/by-name/pi/picom-pijulius/package.nix:10]: The versioning scheme is a bit add. It should be like this per https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming

```suggestion
  version = "8.2-unstable-2024-04-30";
```

File: pkgs/by-name/pi/picom-pijulius/package.nix [hunk 2]
```diff
@@ -2,17 +2,18 @@
...
 
   src = fetchFromGitHub {
     owner = "pijulius";
     repo = "picom";
-    rev = "e7b14886ae644aaa657383f7c4f44be7797fd5f6";
-    hash = "sha256-YQVp5HicO+jbvCYSY+hjDTnXCU6aS3aCvbux6NFcJ/Y=";
+    rev = "083421980af3983a4ca8bd56373303dc0282b395";
```
Comment (JohnRTitor) [inline pkgs/by-name/pi/picom-pijulius/package.nix:15]: Is there a reason we are not using the latest commit here?

File: pkgs/by-name/pi/picom-pijulius/package.nix [hunk 3]
```diff
@@ -29,4 +30,11 @@ picom.overrideAttrs (previousAttrs: {
     homepage = "https://github.com/pijulius/picom";
     maintainers = with lib.maintainers; [ YvesStraten ];
   };
+
+  passthru.updateScript = nix-update-script {
```
Comment (JohnRTitor) [inline pkgs/by-name/pi/picom-pijulius/package.nix:34]: `unstableGitUpdater` would work better here, I believe.

**Review by JohnRTitor — DISMISSED** [review]
Thanks for your contribution, I have left some review comments for you to check.

File: pkgs/by-name/pi/picom-pijulius/package.nix (hunk 1 above)
Comment (YvesStraten) [reply pkgs/by-name/pi/picom-pijulius/package.nix:10]: This comes directly from nix-update, run with the script in maintainers/scripts/update.nix

File: pkgs/by-name/pi/picom-pijulius/package.nix (hunk 2 above)
Comment (YvesStraten) [reply pkgs/by-name/pi/picom-pijulius/package.nix:15]: That is the latest commit, at least according to nix-update

File: pkgs/by-name/pi/picom-pijulius/package.nix (hunk 3 above)
Comment (YvesStraten) [reply pkgs/by-name/pi/picom-pijulius/package.nix:34]: I'll try to use that instead, I couldn't check whether it worked last time as I was on darwin

File: pkgs/by-name/pi/picom-pijulius/package.nix (hunk 1 above)
Comment (JohnRTitor) [reply pkgs/by-name/pi/picom-pijulius/package.nix:10]: You can use `unstableGitUpdater` and run it again! That'll give you the correct version formatting.

**Comment by YvesStraten** [comment]
Thank you very much for your comments, I'll check what's up with nix-update later

**Comment by YvesStraten** [comment]
I have tried, and I am very close, but unfortunately, without the regex, `unstableGitUpdater` fails, with the regex, the name is replaced with a starting `0`. If anyone could help me with this that would be highly appreciated, thank you very much :pray:

**Comment by heisfer** [comment]
@JohnRTitor shouldn't this be `0-unstable-x.x.x` instead for Auto Updater work correctly? 
Also, versioning schema was last used 4 years ago, but repo is still actively being maintained, so I think it should be redundant.

```
  passthru.updateScript = unstableGitUpdater {
    branch = "next";
    hardcodeZeroVersion = true;
  };
```
I also found this package: https://github.com/NixOS/nixpkgs/commits/master/pkgs/development/libraries/cegui/default.nix for the reference.

**Comment by JohnRTitor** [comment]
The last "released" tag is just `vNext` and the tag before that was `8.2`. I do not think if it would be appropriate to hardcode "0" as the version prefix. 

8.2 could work. And we could use `tagFormat` to convert the tag appropriately? https://github.com/NixOS/nixpkgs/blob/ca4f0becf996b69b22598649b0bc9b52ef9f7b5d/pkgs/common-updater/unstable-updater.nix#L20

**Comment by YvesStraten** [comment]
I don't know how to use it, but I will look into it

**Comment by YvesStraten** [comment]
Has been done @JohnRTitor, please review whether the commits are fine, thank you in advance

**Review by JohnRTitor — APPROVED** [review]
LGTM

update script tested and works

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 316783` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>picom-pijulius</li>
  </ul>
</details>

**Pull request merged**